sdk/OpenTap.Sdk.MSBuild/OpenTapImageInstaller.cs

```diff
@@ -17,11 +16,41 @@ internal interface IImageDeployer
 
     internal class DefaultImageDeployer : IImageDeployer
     {
-        public void Install(ImageSpecifier spec, Installation install, CancellationToken cts)
+        static bool IsSystemWide(PackageDef package)
         {
-            if (!spec.Repositories.Any(r => r.ToLower().Contains("https://packages.opentap.io") || r.ToLower().Contains("http://packages.opentap.io")))
-                spec.Repositories.Add("https://packages.opentap.io");
-            spec.MergeAndDeploy(install, cts);
+            return package.Class.Equals("system-wide", StringComparison.OrdinalIgnoreCase);
+        }
+
+        public void Install(ImageSpecifier spec, Installation install, CancellationToken ct)
+        {
+            ImageIdentifier image = null;
+            // As a performance optimization, don't start resolving the image if a package
+            // which is known to be system-wide was requested.
+            var requestedPackages = spec.Packages.Select(p => p.Name).ToArray();
+            var requestedSystemWide = install.GetPackages().Where(IsSystemWide)
```

Comment:
It looks like this is only able to detect the error if the system wide package being installed is already installed and as such does not work in general.

Could we find a better solution?